samples/weather_function/Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM openjdk:8u171-jdk-stretch as builder
+WORKDIR /opt
+ENV KOTLIN_NATIVE_VER "0.7.1"
+RUN wget "https://github.com/JetBrains/kotlin-native/releases/download/v0.7.1/kotlin-native-linux-$KOTLIN_NATIVE_VER.tar.gz"
+RUN tar -xzf kotlin-native-linux-$KOTLIN_NATIVE_VER.tar.gz && rm kotlin-native-linux-$KOTLIN_NATIVE_VER.tar.gz
+RUN apt-get update && apt-get install libcurl4-openssl-dev
+
+ENV KOTLIN_NATIVE_HOME "/opt/kotlin-native-linux-$KOTLIN_NATIVE_VER"
+RUN mkdir -p /app/src/main/kotlin/org/example/weather_func
+RUN mkdir -p /app/lib/cJSON-1.7.7/include/cjson && mkdir -p /app/lib/cJSON-1.7.7/lib/x86_64-linux-gnu
+RUN mkdir -p /app/gradle/wrapper
+WORKDIR /app
+
+COPY gradle/wrapper/gradle-wrapper.properties /app/gradle/wrapper
+COPY gradle/wrapper/gradle-wrapper.jar /app/gradle/wrapper
+COPY gradlew /app
+COPY .konan/ /root/.konan
+COPY *.def /app/
+COPY *.kts /app/
+COPY openweathermap_key.txt /app
+COPY src/ /app/src
+COPY lib/cJSON-1.7.7/include/cjson/cJSON.h /app/lib/cJSON-1.7.7/include/cJSON.h
```

Comment:
This is very verbose and each step will take additional time. I suggest you add the whole folder that you want, since this uses a multi-stage build there is no advantage to picking individual files out.

Reply:
Noticed that the Docker image build takes a while to copy over the Konan cache (does it every time Docker is building a image), and a user would need to compile a basic Linux amd64 program with Kotlin Native **before** the Konan cache can be copied over. Is there a easier way to manage the Konan cache?